React/memory-game/src/components/CardGrid.tsx

```diff
@@ -0,0 +1,44 @@
+import Card from "./Card";
+import styles from "../styles/cardGrid.module.css"
+import IScore from "../interfaces/IScore";
+
+
+export default function CardGrid({ score, scoreCallback}: IScore) {
+
+    const baseUrl = "https://pokeapi.co/api/v2/pokemon/";
+    const pokemonNames = ["gyarados", "ditto", "crobat", "seadra", "electabuzz", "weedle", "golbat", "garchomp", "inteleon", "azelf", "bastiodon", "eternatus"];
+
+    /**shuffle the list of pokemonNames so theyre random */
+    function shuffle(array: string[]) {
+        let currentIndex = array.length;
+      
+        // While there remain elements to shuffle...
+        while (currentIndex != 0) {
+      
+          // Pick a remaining element...
+          let randomIndex = Math.floor(Math.random() * currentIndex);
+          currentIndex--;
+      
+          // And swap it with the current element.
+          [array[currentIndex], array[randomIndex]] = [array[randomIndex], array[currentIndex]];
+        }
+    }
+
+    shuffle(pokemonNames)
+
+    
+
+
+    /** get pokemon card objects from url and name */
+    function getCardForPokemon(api_url: string) {
```

Comment:
You could also just pass name here and concatenate the baseUrl and the name in line, then use just the name as the key